### PR TITLE
🐛Protect against video freezing in stories

### PR DESCRIPTION
**Pull request by calebcordry**
Partial fix for #25954

### CODE REVIEW

File: src/service/video-manager-impl.js [hunk 1]
```diff
@@ -1487,6 +1487,14 @@ export class AnalyticsPercentageTracker {
     }
 
     const duration = video.getDuration();
+    // Handle cases such as livestreams or videos with no duration information is
+    // available, where 1 second is the default duration for some video players
+    // TODO(#25954): Further investigate root cause and remove this protection
+    // if appropriate.
+    if (!duration || isNaN(duration) || duration <= 1) {
```
Comment (alanorozco) [inline src/service/video-manager-impl.js]: Nit: Could you extract a function to keep these two cases consolidated?

```js
/**
 * Handle cases such as livestreams or videos with no duration information is
 * available, where 1 second is the default duration for some video players.
 * @param {?number=} duration
 * @return {boolean}
 */
const isDurationFiniteNonZero = duration => 
  duration && !isNaN(duration) && duration > 1
```

File: test/unit/test-video-analytics-percentage-tracker.js [hunk 2]
```diff
@@ -148,6 +148,30 @@ describes.fakeWin(
...
+          durationStub.returns(invalidDuration);
+
+          const {element} = video;
+
+          setPlayingState(mockEntry, PlayingStates.PLAYING_MANUAL);
+          tracker.start();
+
+          dispatchLoadedMetadata(element);
+
+          expect(mockTimer.delay).to.not.have.been.calledWith(
```
Comment (alanorozco) [inline test/unit/test-video-analytics-percentage-tracker.js]: This should check that this is called with the constant frequency for `paused` state, no?

File: src/service/video-manager-impl.js (hunk 1 above)
Comment (calebcordry) [reply src/service/video-manager-impl.js]: Done.

File: test/unit/test-video-analytics-percentage-tracker.js (hunk 2 above)
Comment (calebcordry) [reply test/unit/test-video-analytics-percentage-tracker.js]: Not sure I understand fully. This just ensures the `delay()` method  is never passed the "invalid" values that could cause the infinite loop. Is there something else you are looking for?

Comment (alanorozco) [reply test/unit/test-video-analytics-percentage-tracker.js]: The invalid values are never passed through in any case. I mean that it should check the call on the conditional:

```js
timer.delay(calculateAgain, PERCENTAGE_FREQUENCY_WHEN_PAUSED_MS);
```

Comment (calebcordry) [reply test/unit/test-video-analytics-percentage-tracker.js]: Done.

**Pull request merged**